tls/s2n_client_cert.c

```diff
@@ -130,8 +131,15 @@ int s2n_client_cert_recv(struct s2n_connection *conn)
 
     /* Determine the Cert Type, Verify the Cert, and extract the Public Key */
     s2n_pkey_type pkey_type = S2N_PKEY_TYPE_UNKNOWN;
-    POSIX_GUARD_RESULT(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
-            cert_chain_size, &pkey_type, &public_key));
+    s2n_result result = s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
+            cert_chain_size, &pkey_type, &public_key);
+    if (s2n_result_is_error(result)) {
```

Comment:
There isn't really a need to reset the stuffer position outside of an async blocked error, so it might be best to gate this branch on that:
```suggestion
    if (s2n_result_is_error(result) && s2n_errno == S2N_ERR_ASYNC_BLOCKED) {
```

This will also require moving `POSIX_GUARD_RESULT(result);` out of this branch.